### PR TITLE
Update HPA Bestfitting Densenet

### PR DESCRIPTION
**Pull request by bioimageiobot**
This is an automatic PR created by the @bioimageiobot regarding changes to the resource item 10.5281/zenodo.5910163.
The following version(s) will be added:
  - 10.5281/zenodo.5910164

You can find their corresponding RDF files here:
  - https://zenodo.org/api/files/5be7a94a-0f23-4658-bee3-d160c6eb1c83/rdf.yaml

The corresponding deployment preview (will be) available at: [gh-pages](/bioimage-io/collection-bioimage-io/tree/gh-pages/resources/10.5281/zenodo.5910163)

Please review the changes and make sure the new item or version(s) pass the following check list:
- [x] Passed the BioImage.IO Spec validator (static validation);
- [ ] Passed the BioImage.IO CI tests (dynamic validations);
- [ ] The meta information for the RDF item is complete;
  - [ ] The tags is complete and describes the model;
  - [ ] Naming is intuitive and descriptive, example:  Multi-Organ Nucleus Segmentation (StarDist 2D);
  - [ ] Authors are provided;
  - [ ] Documentation is complete;
    * For models, include an overview, describe how the model is trained, what is the training data, how to use the model, how to validate the results and list the references. TODO: Model documentation template.
- [ ] Approved by at least one of the BioImage.IO admin team member.

Maintainers: none specified

Note: If you updated or re-uploaded another version for the current item on Zenodo, this PR won't be changed automatically. To proceed, you can do the following:
 1. Edit the current resource.yaml and mark the top-level `status` field as `accepted`, but change the `status` under the current version to `blocked`.
 2. Merge this PR for now, wait until the CI on the main branch to complete. It should detect the new versions and create another PR for the new version. You can take it from there instead.

### CODE REVIEW

**Comment by oeway** [comment]
@FynnBe Could you please look into this validation error, is it because of the recent change to the rdf_source field, it seems that it try to load the rdf from gh-pages which is not ready yet.

**Comment by FynnBe** [comment]
I see... we wanted to deploy also pending versions to gh-pages (just not including them in the generated collection rdf.yaml), but this is not happening yet. I'll merge the two main workflows `auto_update_main.yaml` (where we create new auto PRs) and `build_collection_main.yaml` (where we deploy resource versions to gh-pages for accepted versions) and make sure we deploy all non-blocked versions...

**Comment by oeway** [comment]
@FynnBe Sure we need to fix that, but would it be easier we figure out the right rdf_source for running the validation, in the collection/rdf.yaml in the pr, we have the rdf_source point to zenodo, however, somehow the validation script use the gh-pages version. I think we should always use the zenodo one for running the validation. Once merged, the website will use the gh-pages cached version (which contains overrieded fields).

**Comment by FynnBe** [comment]
> @FynnBe Sure we need to fix that, but would it be easier we figure out the right rdf_source for running the validation, in the collection/rdf.yaml in the pr, we have the rdf_source point to zenodo, however, somehow the validation script use the gh-pages version. I think we should always use the zenodo one for running the validation. Once merged, the website will use the gh-pages cached version (which contains overrieded fields).

because we allow to override fields we should always validate the potentially overridden one on gh-pages...

**Comment by oeway** [comment]
> 

True, but can we generate a rdf_source on the fly so we don't rely on the push action to gh-pages during in the PR?

**Comment by FynnBe** [comment]
just merging the above mentioned workflows doesn't cut it though...

**Comment by oeway** [comment]
> 

Well, it still means the validation script will try to load from gh-pages, right? Which can be problematic for local testing too, no? For testing it locally, we don't want to push to gh-pages and pull it back to run validation.

**Comment by oeway** [comment]
> 

I think I get the issue now, let's do as you suggested for now.

**Comment by FynnBe** [comment]
> Well, it still means the validation script will try to load from gh-pages, right? Which can be problematic for local testing too, no?

I think it is the easiest to always pull from gh-pages. Initially it will just hold the plain copy from source without any updates. this can be put there before opening a PR. For local testing I don't see the problem because all known sources will be on gh-pages (this only rules out testing with unknown new resources)

File: collection/10.5281/zenodo.5910163/resource.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+doi: 10.5281/zenodo.5910163
+id: 10.5281/zenodo.5910163
+owners: [47882]
+status: accepted
+type: model
+versions:
+- {created: '2022-01-27 10:28:20.092577', doi: 10.5281/zenodo.5910164, name: HPA Bestfitting
```
Comment (oeway) [inline collection/10.5281/zenodo.5910163/resource.yaml:7]: A minor thing, can we format the version yaml so we have one field per line?

Comment (FynnBe) [reply collection/10.5281/zenodo.5910163/resource.yaml:7]: I'll look into it

**Comment by oeway** [comment]
@FynnBe No pressure but it seems all the PR are failing because of this issue we discussed above, will you give it a try?

**Comment by FynnBe** [comment]
> @FynnBe No pressure but it seems all the PR are failing because of this issue we discussed above, will you give it a try?

yes, I'll fix it. sorry for the delay

**Comment by FynnBe** [comment]
I'm still working on it: https://github.com/bioimage-io/collection-bioimage-io/pull/268

**Comment by constantinpape** [comment]
@oeway It looks like this still has the issue with the preprocessing:
```
{'validation_summary_onnx': {'error': 'arguments without labels along '
                                      "dimension 'c' cannot be aligned because "
                                      'they have different dimension sizes: '
                                      '{1, 4}',
                             'name': 'reproduced test outputs from test inputs',
                             'traceback': ['  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/bioimageio/core/resource_tests.py", '
                                           'line 130, in test_resource\n'
                                           '    results = '
                                           'predict(prediction_pipeline, '
                                           'inputs)\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/bioimageio/core/prediction.py", '
                                           'line 261, in predict\n'
                                           '    return '
                                           'prediction_pipeline.forward(*tagged_data)\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/bioimageio/core/prediction_pipeline/_prediction_pipeline.py", '
                                           'line 128, in forward\n'
                                           '    preprocessed, sample_stats = '
                                           'self._processing.apply_preprocessing(*input_tensors)\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/bioimageio/core/prediction_pipeline/_combined_processing.py", '
                                           'line 105, in apply_preprocessing\n'
                                           '    tensors[proc.tensor_name] = '
                                           'proc.apply(tensors[proc.tensor_name])\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/bioimageio/core/prediction_pipeline/_processing.py", '
                                           'line 149, in apply\n'
                                           '    return ensure_dtype(tensor * '
                                           'gain + offset, dtype="float32")\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/xarray/core/_typed_ops.py", '
                                           'line 212, in __mul__\n'
                                           '    return self._binary_op(other, '
                                           'operator.mul)\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/xarray/core/dataarray.py", '
                                           'line 3081, in _binary_op\n'
                                           '    self, other = align(self, '
                                           'other, join=align_type, '
                                           'copy=False)\n',
                                           '  File '
                                           '"/usr/share/miniconda/envs/10.5281zenodo.5910164/lib/python3.10/site-packages/xarray/core/alignment.py", '
                                           'line 348, in align\n'
                                           '    raise ValueError(\n']}}
```

**Comment by oeway** [comment]
@FynnBe See here: https://github.com/bioimage-io/collection-bioimage-io/runs/5029116108?check_suite_focus=true This is the action trigger by my close action above, however, it doesn't give me a new PR.

**Comment by FynnBe** [comment]
> @FynnBe See here: https://github.com/bioimage-io/collection-bioimage-io/runs/5029116108?check_suite_focus=true This is the action trigger by my close action above, however, it doesn't give me a new PR.

very surprising. I'll look into it on my fork...

**Pull request merged**